src/widgets/scrollbar.rs

```diff
@@ -0,0 +1,208 @@
+use crate::{
+    buffer::Buffer,
+    layout::Rect,
+    style::Style,
+    symbols::{block::FULL, line::DOUBLE_VERTICAL},
+    widgets::Widget,
+};
+
+/// Scrollbar widget for tui-rs library.
+///
+/// This widget can be used to display a scrollbar in a terminal user interface.
+#[derive(Default, Debug, Clone)]
+pub struct Scrollbar<'a> {
+    scroll: f64,
+    style: Style,
+    symbol: &'a str,
+    bar_style: Style,
+    bar_symbol: &'a str,
+}
+
+impl<'a> Scrollbar<'a> {
+    /// Creates a new scrollbar with default styles and symbols.
+    pub fn new(scroll: f64) -> Self {
+        Self {
+            scroll,
+            style: Style::default(),
+            symbol: FULL,
+            bar_style: Style::default(),
+            bar_symbol: DOUBLE_VERTICAL,
+        }
+    }
+
+    /// Sets the scroll position of the scrollbar.
+    pub fn scroll(mut self, scroll: f64) -> Self {
+        if scroll < 0.0 {
+            self.scroll = 0.0;
+        } else {
+            self.scroll = scroll;
+        }
+        self
+    }
+
+    /// Sets the symbol that represents the scroller.
+    pub fn symbol(mut self, symbol: &'a str) -> Self {
+        self.symbol = symbol;
+        self
+    }
+
+    /// Sets the style that is used for the scroller.
+    pub fn style(mut self, style: Style) -> Self {
+        self.style = style;
+        self
+    }
+
+    /// Sets the symbol that represents the bar of the scrollbar.
+    pub fn bar_symbol(mut self, bar_symbol: &'a str) -> Self {
+        self.bar_symbol = bar_symbol;
+        self
+    }
+
+    /// Sets the style that is used for the bar of the scrollbar.
+    pub fn bar_style(mut self, bar_style: Style) -> Self {
+        self.bar_style = bar_style;
+        self
+    }
+}
+
+impl<'a> Widget for Scrollbar<'a> {
+    /// Renders the scrollbar.
+    ///
+    /// The scrollbar is drawn with a fixed width on the right edge of the given area.
+    fn render(self, area: Rect, buf: &mut Buffer) {
+        // If the area is too small, we don't draw the scrollbar.
+        if area.height <= 2 {
```

Comment:
nit: use a constant to avoid writing a comment (e.g. `MIN_HEIGHT`)

Reply:
Do you mean writing it like this?

```
        const MIN_HEIGHT: u16 = 0;

        if area.height == MIN_HEIGHT {
            return;
        }
```

Reply:
More try to explain why the min height is 2 in a constant - here's it's because we're subtracting the top and the bottom row and not rendering them. Something like
```rust
// subtract space for top and bottom
let Some(bar_height) = area.height.checked_sub(2) or else { return; };
```

I think this has disappeared in the current iteration though (which checks for 0 - which is obvious and fine).